pkg/kademlia/kademlia.go

```diff
@@ -57,14 +58,15 @@ type Kademlia struct {
 }
 
 // NewService returns a newly configured Kademlia instance
-func NewService(log *zap.Logger, self pb.Node, bootstrapNodes []pb.Node, transport transport.Client, alpha int, rt *RoutingTable) (*Kademlia, error) {
+func NewService(log *zap.Logger, self pb.Node, transport transport.Client, rt *RoutingTable, config Config) (*Kademlia, error) {
 	k := &Kademlia{
 		log:              log,
-		alpha:            alpha,
+		alpha:            config.Alpha,
 		routingTable:     rt,
-		bootstrapNodes:   bootstrapNodes,
+		bootstrapNodes:   config.BootstrapNodes(),
 		dialer:           NewDialer(log.Named("dialer"), transport),
 		refreshThreshold: int64(time.Minute),
+		queryTimeout:     config.QueryTimeout,
```

Comment:
This timeout seems more about the endpoint and less about the service.

Reply:
how should NewEndpoint access this value then besides via the kademlia config?

Reply:
You can pass the timeout as an argument to `NewEndpoint` flag rather than kademlia service.

Reply:
unless I add an Endpoint config?

Reply:
but I'm not sure how I would let testplanet have access to an Endpoint config.. should the endpoint config be passed to NewService?

Reply:
never mind, I see! I think I'll just keep it as an argument.